### PR TITLE
Enable ASAN for tests.

### PR DESCRIPTION
**Pull request by ademuri**
Fix all detected memory leaks.

### CODE REVIEW

File: src/generic/test/FakeNetwork.hpp [hunk 1]
```diff
@@ -39,8 +39,9 @@ class FakeNetwork {
 
  private:
   int packet_loss = 0;
-  FakeRadio radios[kNumNodes];
+  FakeRadio *radios[kNumNodes];
```
Comment (wmmiii) [inline src/generic/test/FakeNetwork.hpp]: Why does this need to be a pointer to an array?

**Comment by wmmiii** [comment]
How did you find all these leaks?

**Review by wmmiii — APPROVED** [review]
LGTM, can we add a CI check too in a follow-up?

**Comment by ademuri** [comment]
CI will check this - I added the `-fsanitize=address` flag to the CMake config, which makes the test binary run these checks.

File: src/generic/test/FakeNetwork.hpp (hunk 1 above)
Comment (ademuri) [reply src/generic/test/FakeNetwork.hpp]: It doesn't, reverted.

**Pull request merged**